crd-generator/api-v2/src/main/java/io/fabric8/crdv2/generator/CRDGenerator.java

```diff
@@ -201,22 +209,40 @@ public CRDGenerationInfo detailedGenerate() {
 
         if (parallel) {
           handlers.values().stream().map(h -> ForkJoinPool.commonPool().submit(() -> h.handle(info, context.forkContext())))
-              .collect(Collectors.toList()).stream().forEach(ForkJoinTask::join);
+              .collect(Collectors.toList())
+              .forEach(ForkJoinTask::join);
         } else {
-          handlers.values().stream().forEach(h -> h.handle(info, context.forkContext()));
+          handlers.values().forEach(h -> h.handle(info, context.forkContext()));
         }
       }
     }
 
     final CRDGenerationInfo crdGenerationInfo = new CRDGenerationInfo();
     handlers.values().stream().flatMap(AbstractCustomResourceHandler::finish)
-        .forEach(crd -> emitCrd(crd.getKey(), crd.getValue(), crdGenerationInfo));
+        .forEach(crd -> emitCrd(crd.getKey(), crd.getValue(), crdGenerationInfo, processor));
     return crdGenerationInfo;
   }
 
+  public CRDGenerationInfo detailedGenerate() {
+    return detailedGenerate(nullProcessor);
+  }
+
+  /**
+   * @deprecated use {@link #emitCrd(HasMetadata, Set, CRDGenerationInfo, CRDPostProcessor)} instead
```

Comment:
The new method is marked protected, users won't be able to transition.

On the other hand, why don't we make CRDPostProcessor a member variable (just as `output`, `minQuotes`, etc.) that defaults to the `nullProcessor` -> `withCrdPostProcessor()`.
This way the method signature won't keep growing or changing.

Reply:
Actually, that `emitCRD` method should probably not have been exposed in the first place and I don't think anyone is using it but the suggested change is a good one. Thank you!